arangod/Network/NetworkFeature.cpp

```diff
@@ -327,15 +353,116 @@ void NetworkFeature::sendRequest(network::ConnectionPool& pool,
+             "endpoint:"
+          << endpoint << ", request ptr: " << (void*)req.get()
+          << ", response ptr: " << (void*)res.get()
+          << ", error: " << uint16_t(err);
+      TRI_ASSERT(false);  // should never happen
+    } else if (req->timeAsyncWrite().time_since_epoch().count() == 0) {
+      if (err != fuerte::Error::ConnectionClosed) {
+        // If the connection is already in closed state, we do not even
+        // queue the item, likewise, if the queue is full, but we do want
+        // to see this, if it happens.
+        LOG_TOPIC("effc1", WARN, Logger::COMMUNICATION)
+            << "Request was queued in fuerte, but the sending never started, "
+               "endpoint:"
+            << endpoint << ", request ptr: " << (void*)req.get()
+            << ", response ptr: " << (void*)res.get()
+            << ", error: " << uint16_t(err);
+      }
+    } else {
+      auto dur = req->timeAsyncWrite() - req->timeReceived();
+      if (dur > std::chrono::seconds(1)) {
+        LOG_TOPIC("effc2", WARN, Logger::COMMUNICATION)
+            << "Time to dequeue request to " << endpoint << ": "
+            << std::chrono::duration_cast<std::chrono::duration<double>>(dur)
+                   .count()
+            << " seconds, endpoint: " << endpoint
+            << ", request ptr: " << (void*)req.get()
+            << ", response ptr: " << (void*)res.get()
+            << ", error: " << uint16_t(err);
+        _delayedDequeues.count();
+      }
+      if (req->timeSent().time_since_epoch().count() == 0) {
+        LOG_TOPIC("effc3", WARN, Logger::COMMUNICATION)
+            << "Time to dequeue request to " << endpoint << ": "
+            << std::chrono::duration_cast<std::chrono::duration<double>>(dur)
+                   .count()
+            << " seconds, however, the sending has not yet finished so far"
+            << ", endpoint: " << endpoint
+            << ", request ptr: " << (void*)req.get()
+            << ", response ptr: " << (void*)res.get()
+            << ", error: " << uint16_t(err);
+        _unfinishedSends.count();
+      } else {
+        dur = req->timeSent() - req->timeAsyncWrite();
+        _sendDurations.count(dur.count());
+        if (dur > std::chrono::seconds(3)) {
+          LOG_TOPIC("effc4", WARN, Logger::COMMUNICATION)
+              << "Time to send request to " << endpoint << ": "
+              << std::chrono::duration_cast<std::chrono::duration<double>>(dur)
+                     .count()
+              << " seconds, endpoint: " << endpoint
+              << ", request ptr: " << (void*)req.get()
+              << ", response ptr: " << (void*)res.get()
+              << ", error: " << uint16_t(err);
+          _delayedSends.count();
+        }
+        dur = std::chrono::steady_clock::now() - req->timeSent();
+        if (dur > std::chrono::seconds(61)) {
+          LOG_TOPIC("effc5", WARN, Logger::COMMUNICATION)
```

Comment:
I think logging a warning here is too harsh, because it will log a warning for requests that are known to be taking long (e.g. requests for heavyweight AQL queries etc.). I think we should downgrade the log level here from warning to info, because it *can* happen.